src/Moq/SetupCollection.cs

```diff
@@ -101,6 +105,7 @@ public void Clear()
 			lock (this.setups)
 			{
 				this.setups.Clear();
+				this.activeInvocationShapes.Clear();
```

Comment:
Just a remark for posteriority: in theory, we should also filter out removed setups in `RemoveAllPropertyAccessorSetups`. Not doing means that the first time a property setup is recreated after a call to `mock.SetupAllProperties()`, there may be a superfluous call to `MarkOverriddenSetups`. This is an edge case however, and the overhead of the additional `MarkOverriddenSetups` in this case is probably quite small, so let's just ignore this for now.